### PR TITLE
Added check if the server is started or not.

### PR DESCRIPTION
**Pull request by SeppPenner**
Fixes https://github.com/chkr1011/MQTTnet/issues/846.

### CODE REVIEW

**Comment by SeppPenner** [comment]
I'm not sure if this is a good idea, it was just a proposal and easy to implement...

File: Source/MQTTnet/Server/MqttServer.cs [hunk 1]
```diff
@@ -190,6 +190,8 @@ public Task ClearRetainedApplicationMessagesAsync()
             return _retainedMessagesManager?.ClearMessagesAsync();
         }
 
+        public bool IsStarted => _cancellationTokenSource != null;
```
Comment (chkr1011) [inline Source/MQTTnet/Server/MqttServer.cs]: Please also move this to the server interface because you have to cast the object from the MQTT factory.

**Comment by chkr1011** [comment]
Please add an entry to the nuspec file which describes this change.

**Comment by SeppPenner** [comment]
I changed the pull request.

File: Source/MQTTnet/Server/MqttServer.cs (hunk 1 above)
Comment (chkr1011) [reply Source/MQTTnet/Server/MqttServer.cs:193]: Also please move the property to the property section of the file. Be best is at corresponding other properties for a similar purpose.

And please merge master and fix conflicts.

**Pull request merged**